Sources/Plasma/NucleusLib/pnAsyncCoreExe/Private/Nt/pnAceNt.cpp

```diff
@@ -162,23 +127,13 @@ static void INtOpDispatch (
             // critical section to do so. This is a big win because a single operation
             // that takes a long time to complete can backlog a long list of completed ops.
             for (;;) {
-                // wake up any other threads waiting on this event
-                CNtWaitHandle * signalComplete = op->signalComplete;
-                op->signalComplete = nullptr;
-
                 // since this operation is at the head of the list we can complete it
                 if (op->asyncId && !++ntObj->nextCompleteSequence)
                     ++ntObj->nextCompleteSequence;
                 Operation * next = ntObj->opList.Next(op);
                 ntObj->opList.Delete(op);
                 op = next;
 
-                // set event *after* operation is complete
-                if (signalComplete) {
-                    signalComplete->SignalObject();
-                    signalComplete->UnRef();
-                }
-
```

Comment:
I'm having a trouble understanding how we can remove this wait, mostly because I'm not 100% on why it's here in the first place. That being said, it looks like only `kOpSocketWrite` will get here (a synchronous-ish write) but any `kOpQueuedSocketWrite` posts a `kOpSocketWrite` in `INtSocketOpCompleteQueuedSocketWrite()`. The event seems to only be waited on when the operation is destroyed, though.

Reply:
We're not actually removing the wait, we're removing the `signalComplete` member from `op` entirely, since it was never set to anything other than `nullptr`.  Since it was never set, the local variable would always be `nullptr` too, meaning this if statement was always false.